File: my-workspace/libs/hiking-ui/trip-planning/feature/src/lib/models/search-trip.model.ts
export class SearchTripModel {
  startDate: Date;
  endDate: Date;
  friendsOnly: boolean;
  keywords: string[];

  public constructor() {
    this.friendsOnly = false;
    this.keywords = [];
  }

  clearSearch() {
    this.startDate = null;
    this.endDate = null;
    this.friendsOnly = false;
    this.keywords = [];
  }
}

File: TripService/TripService/Models/ApiModels/SearchTripModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TripService.Models.ApiModels
{
    public class SearchTripModel
    {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string[] Keywords { get; set; }
        public bool FriendsOnly { get; set; }
    }
}
